internal/dockerfile/dockerfile.go

```diff
@@ -122,6 +123,55 @@ func (d Dockerfile) ValidateBaseDockerfile() error {
 	return err
 }
 
+// DeriveMounts finds ADD statements in a Dockerfile and turns them into Tilt model.Mounts.
+// Relative paths in an ADD statement are relative to the build context (passed as an arg)
+// and will appear in the Mount as an abs path.
+func (d Dockerfile) DeriveMounts(context string) ([]model.Mount, error) {
+	result, err := parser.Parse(bytes.NewBufferString(string(d)))
+	if err != nil {
+		return nil, nil
+	}
+
+	var nodes []*parser.Node
+	err = traverse(result.AST, func(node *parser.Node) error {
+		switch strings.ToUpper(node.Value) {
+		case "ADD", "COPY":
+			nodes = append(nodes, node)
+		}
+		return nil
+	})
+
+	mounts := make([]model.Mount, len(nodes))
+	for i, n := range nodes {
+		m, err := nodeToMount(n, context)
+		if err != nil {
+			return nil, err
+		}
+		mounts[i] = m
+	}
+	return mounts, nil
+}
+
+func nodeToMount(node *parser.Node, context string) (model.Mount, error) {
+	cmd := strings.ToUpper(node.Value)
+	if !(cmd == "ADD" || cmd == "COPY") {
+		return model.Mount{}, fmt.Errorf("nodeToMounts works on ADD/COPY nodes; got '%s'", cmd)
+	}
+
+	srcNode := node.Next
+	dstNode := srcNode.Next
+
+	src := srcNode.Value
+	if !path.IsAbs(src) {
+		src = path.Join(context, src)
```

Comment:
this should use `filepath`. `path` is for urls

Reply:
😳